ui/lib/kubernetes/addon/components/page/credentials.hbs

```diff
@@ -0,0 +1,144 @@
+          <LinkTo @route="roles">roles</LinkTo>
+        </li>
+        <li>
+          <span class="sep">/</span>
+          <LinkTo @route="roles.role.details">{{@role.name}}</LinkTo>
+        </li>
+        <li>
+          <span class="sep">/</span>
+          <span>credentials</span>
+        </li>
+      </ul>
+    </nav>
+  </p.top>
+  <p.levelLeft>
+    <h1 class="title is-3 has-bottom-margin-2" data-test-credentials-header>
+      {{if this.credentials "Credentials" "Generate credentials"}}
+    </h1>
+  </p.levelLeft>
+</PageHeader>
+
+{{#if this.credentials}}
+  <div class="box is-sideless is-fullwidth is-marginless has-bottom-padding-l" data-test-credentials-details>
+    <AlertBanner
+      @class="is-marginless"
+      @type="warning"
+      @title="Warning"
+      @message="You won't be able to access these credentials later, so please copy them now."
+    />
+    <InfoTableRow @label="Service account token">
+      <MaskedInput
+        @value={{this.credentials.service_account_token}}
+        @name="Service Account Token"
+        @displayOnly={{true}}
+        @allowCopy={{true}}
+      />
+    </InfoTableRow>
+
+    <InfoTableRow @label="Namespace">
+      <CopyButton
+        class="button is-transparent is-flex-v-centered"
+        @clipboardText={{this.credentials.service_account_namespace}}
+        @buttonType="button"
+        @success={{action (set-flash-message "Namespace copied")}}
+      >
+        <Icon class="has-text-grey-light" @name="clipboard-copy" aria-label="Copy" />
+        <span class="has-font-weight-normal title is-7">{{this.credentials.service_account_namespace}}</span>
+      </CopyButton>
+    </InfoTableRow>
+
+    <InfoTableRow @label="Service account name">
+      <CopyButton
+        class="button is-transparent is-flex-v-centered"
+        @clipboardText={{this.credentials.service_account_name}}
+        @buttonType="button"
+        @success={{action (set-flash-message "Service account name copied")}}
+      >
+        <Icon class="has-text-grey-light" @name="clipboard-copy" aria-label="Copy" />
+        <span class="has-font-weight-normal title is-7">{{this.credentials.service_account_name}}</span>
+      </CopyButton>
+    </InfoTableRow>
+
+    <InfoTableRow @label="Lease expiry" @value={{date-format this.leaseExpiry "MMMM do yyyy, h:mm:ss a"}} />
+    <InfoTableRow @label="lease_id" @value={{this.credentials.lease_id}} />
+    <InfoTableRow />
+  </div>
+
+  <div class="has-top-margin-l">
+    <button class="button is-primary" type="button" {{on "click" this.cancel}}>
+      Done
+    </button>
+  </div>
+{{else}}
+  <div class="box is-sideless is-fullwidth is-marginless" data-test-generate-credentials>
+    <p>This will generate credentials using the role <span class="is-font-mono">rolename</span>.</p>
+
+    {{#if this.error}}
+      <MessageError class="has-top-margin-l" @errorMessage={{this.error}} />
+    {{/if}}
+
+    <div class="field has-top-margin-l">
```

Comment:
If these fields were wrapped in a `form` element and the `Generate Credentials` button type was changed to `submit` I think this would be a better experience from an accessibility standpoint, at least having the `Enter` key submit the form.